trie/rlphacks/bytes.go

```diff
@@ -37,7 +39,7 @@ func (b RlpEncodedBytes) DoubleRLPLen() int {
 
 func encodeBytesAsRlpToWriter(source []byte, w io.Writer, prefixGenFunc func([]byte, int, int) int, prefixBufferSize uint) error {
 	// > 1 byte, write a prefix or prefixes first
-	if len(source) > 1 || (len(source) == 1 && source[0] >= 0x80) {
+	if len(source) > 1 || (len(source) == 1 && source[0] >= rlp.EmptyStringCode) {
```

Comment:
Strictly speaking, in this particular case, 0x80 is not used as the encoding of the empty string. It is another constant of the RLP algorithm, which happens to be the same value as the encoding of the empty string.

Reply:
It is related though, otherwise how would they know if that is a prefix or not, so I understand why it is there :) But I can add a comment about that too.